src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -0,0 +1,84 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import java.io.FileReader
+import java.util.UUID
+
+import akka.actor.ActorSystem
+import com.google.api.client.auth.oauth2.Credential
+import com.google.api.client.googleapis.auth.oauth2.{GoogleAuthorizationCodeFlow, GoogleClientSecrets}
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport
+import com.google.api.client.json.jackson2.JacksonFactory
+import com.google.api.client.util.store.DataStoreFactory
+import com.google.api.services.storage.Storage
+import com.google.api.services.storage.model.Bucket
+import org.broadinstitute.dsde.rawls.RawlsException
+import scala.collection.JavaConversions._   // Seq[String] -> Collection<String>
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import spray.client.pipelining._
+
+class HttpGoogleCloudStorageDAO(clientSecretsFile: String, dataStoreFactory: DataStoreFactory, ourBaseURL: String)(implicit system: ActorSystem) extends GoogleCloudStorageDAO {
+  // modify these if we need more granular access in the future
+  val gcsFullControl = "https://www.googleapis.com/auth/devstorage.full_control"
+  val computeFullControl = "https://www.googleapis.com/auth/compute"
+  val scopes = Seq(gcsFullControl,computeFullControl)
+
+  lazy val httpTransport = GoogleNetHttpTransport.newTrustedTransport
+  lazy val jsonFactory = JacksonFactory.getDefaultInstance
+  lazy val clientSecrets = GoogleClientSecrets.load(jsonFactory, new FileReader(clientSecretsFile))
+
+  lazy val flow = new GoogleAuthorizationCodeFlow.Builder(httpTransport, jsonFactory, clientSecrets, scopes).setDataStoreFactory(dataStoreFactory).build()
+
+  override def getOurRedirectURI(callbackPath: String): String = s"${ourBaseURL}/${callbackPath}"
+
+  override def getGoogleRedirectURI(userId: String, callbackPath: String): String = {
+    flow.newAuthorizationUrl()
+     .setRedirectUri(getOurRedirectURI(callbackPath))
+     .setState(userId)
+     .setAccessType("offline")   // enables token refresh
+     .build()
+  }
+
+  override def storeUser(userId: String, authCode: String, state: String, callbackPath: String): Unit = {
+    if ( userId != state ) throw new RawlsException("Whoa.  We seem to be dealing with the wrong user.")
+    val tokenResponse = flow.newTokenRequest(authCode)
+      .setRedirectUri(getOurRedirectURI(callbackPath))
+      .execute()
+
+    flow.createAndStoreCredential(tokenResponse, userId)
+  }
+
+  override def createBucket(userId: String, projectId: String): String = {
+    val credential = getCredential(userId)
+    val storage = new Storage.Builder(httpTransport, jsonFactory, credential).build()
+    val bucketName = "CloudPilot-"+UUID.randomUUID
```

Comment:
FireCloud, not CloudPilot and I think it would be helpful to have the workspace name in the bucket name.